File: CS-Examples/21_Security/ProtectWithEditableRange.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace ProtectWithEditableRange
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Load an existing Excel document from file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ProtectWithEditableRange.xlsx");

            // Get the first worksheet from the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Define the specified ranges that allow users to edit while the sheet is protected
            sheet.AddAllowEditRange("EditableRanges", sheet.Range["B4:E12"]);

            // Protect the worksheet with a password
            sheet.Protect("TestPassword", SheetProtectionType.All);

            // Specify the output filename for the workbook
            String result = "ProtectWithEditableRange_result.xlsx";

            // Save the modified workbook to a file
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer(result);
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        private void btnAbout_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VB-Examples/21_Security/ProtectWithEditableRange.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace ProtectWithEditableRange
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ProtectWithEditableRange.xlsx")

            ' Get the first worksheet from the Workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Add an editable range named "EditableRanges" for the specified range B4:E12 in the worksheet
            sheet.AddAllowEditRange("EditableRanges", sheet.Range("B4:E12"))

            ' Protect the worksheet with the specified password and enable all protection options
            sheet.Protect("TestPassword", SheetProtectionType.All)

            ' Specify the name of the resulting Excel file after protecting with editable range
            Dim result As String = "ProtectWithEditableRange_result.xlsx"

            ' Save the Workbook to the specified path in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnAbout_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnAbout.Click
			Close()
		End Sub
	End Class
End Namespace
